app/src/main/java/com/example/pix/login/LoginActivity.java

```diff
@@ -25,21 +31,37 @@
     private boolean loggedIn = false;
     private boolean authenticated = false;
     private static SpotifyAppRemote mSpotifyAppRemote;
+    private final static int RECORD_VIDEO = 100;
 
     // We can access the Spotify Remote in later Activities
     public static SpotifyAppRemote getmSpotifyAppRemote() {
         return mSpotifyAppRemote;
     }
 
-    // When one of (spotify, parse) is done, check if both are done
     private void checkIfDone() {
         if (loggedIn && authenticated) {
-            Intent i = new Intent(LoginActivity.this, HomeActivity.class);
-            startActivity(i);
-            finish();
+            // If we are missing a permission, request it
+            if (ActivityCompat.checkSelfPermission(this, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED
+                    || ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED
+                    || ActivityCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED
+                    || ActivityCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED
```

Comment:
I forgot these last two permissions and later realized they are needed and are why I couldn't change the User's profile picture.